lib/avrora/storage/registry.ex

```diff
@@ -93,20 +93,67 @@ defmodule Avrora.Storage.Registry do
   @spec configured?() :: true | false
   def configured?, do: !is_nil(Config.registry_url())
 
+  defp handle_http_client(path, http_client_function) do
```

Comment:
This is a comment for the entire file: I think this module becomes too complex for a basic-auth feature and should be somehow simplified. Maybe when you will scope out unnecessary functionality we can do one more round here